src/service/resources-impl.js

```diff
@@ -332,6 +332,27 @@ export class Resources {
   }
 
   /**
+   * Returns a promise to the layoutBox for the element. If the element is
+   * resource-backed then makes use of the resource layoutBox, otherwise
+   * measures the element directly.
+   * @param {!Element} element
+   * @return {!Promise<!../layout-rect.LayoutRectDef>}
+   */
+  getElementLayoutBox(element) {
```

Comment:
The only nuance will be (actually _is_) - the distance between two elements can only be really calculated when they are in the same layer - otherwise it'd be mostly meaningless. We could consider some form of multi-dimensional vector, but that's probably overkill.